Real-Dev-Squad/website-dashboard
# UI/UX Improvements for the Extension Request Feature

### Developer Name: Joy Gupta

### PR Number(s): 595

### UI Fixes/Improvements:
- [x] Form values persist even after editing, even if the changes are not saved.
- [x] The scrollbar is hidden on MacOS for audit logs.
- [x] Font family varies for the reason that SU clicks on the edit button for an extension request.
- [x] Make UI responsive for longer titles
- [x] newETA is always changing, even if you haven't changed it.
- [x] The clock icon on the extension request card is displaying NaN hrs/week.
- [x] Audit logs should be rendered in real time.
- [x] Auto close calendar after user input.
~~Validate if the same 24-hour time format is being used in the calendar for every use case.~~

### Issue Ticket Number:
- https://github.com/Real-Dev-Squad/website-dashboard/issues/589

### Backend Changes

### Frontend Changes

### Is Under Feature Flag

### Database changes: 

### Breaking changes (If your feature is breaking/missinsomething please mention pending tickets): 

### Deployment notes:
NA

### Tested in local

### QA Instructions, Screenshots, Recordings

Run `yarn test` and there you can see test case running.
![image](https://github.com/Real-Dev-Squad/website-dashboard/assets/56365512/e54f5e80-a036-43c1-9af0-0cdefc48800a)


### Working Proof
1. Form values should be discarded once the user cancels the changes.

https://github.com/Real-Dev-Squad/website-dashboard/assets/56365512/e40e13f4-cf18-41d1-ae34-b62e0e07486b


2. Scrollbar should persist all the time if overflow is set to scroll

https://github.com/Real-Dev-Squad/website-dashboard/assets/56365512/c7996dc3-e24c-4fe7-bc6b-cf6f708cec8a


3. Font family & font size vary for the reason that SU clicks on the edit button for an extension request.
4. Make UI responsive for longer titles

https://github.com/Real-Dev-Squad/website-dashboard/assets/56365512/4120d77c-1828-4847-ba90-02c2de528cf7

5. newEndsOn is always changing, even if you haven't changed it. (To verify check newEndsOn passed in the PATCH request is the same as provided by the server)

https://github.com/Real-Dev-Squad/website-dashboard/assets/56365512/1a2e0c30-e613-4350-ae7e-b919f06dec7f


6. The clock icon on the extension request card is displaying NaN hrs/week

![Screenshot from 2023-10-29 02-46-20](https://github.com/Real-Dev-Squad/website-dashboard/assets/56365512/4bb587c2-dd8d-46fd-9077-5b952d7a921c)


7. Auto close calendar after user input.

[simplescreenrecorder-2023-10-29_02.49.05.webm](https://github.com/Real-Dev-Squad/website-dashboard/assets/56365512/8a8333eb-2d5f-47eb-993f-92cc729a6977)


8. Audit logs should be rendered in real-time.

https://github.com/Real-Dev-Squad/website-dashboard/assets/56365512/19cb4af9-f9cd-4578-b221-c5de698baded
